proto/ibc/applications/fee/v1/fee.proto

```diff
@@ -41,8 +41,22 @@ message IdentifiedPacketFee {
   repeated string relayers       = 4;
 }
 
-// IdentifiedPacketFees contains a list of packet fees
+// PacketFee contains the relayer fee, refund address and an optional list of relayers that are permitted to receive the
+// fee
+message PacketFee {
+  Fee             fee            = 1 [(gogoproto.nullable) = false];
+  string          refund_address = 2 [(gogoproto.moretags) = "yaml:\"refund_address\""];
+  repeated string relayers       = 3;
+}
+
+// PacketFees contains a list of type PacketFee
+message PacketFees {
+  repeated PacketFee packet_fees = 1 [(gogoproto.moretags) = "yaml:\"packet_fees\"", (gogoproto.nullable) = false];
+}
+
+// IdentifiedPacketFees contains a list of type PacketFee and associated PacketId
 message IdentifiedPacketFees {
-  repeated IdentifiedPacketFee packet_fees = 1
-      [(gogoproto.nullable) = false, (gogoproto.moretags) = "yaml:\"packet_fees\""];
-}
\ No newline at end of file
+  ibc.core.channel.v1.PacketId packet_id = 1
+      [(gogoproto.nullable) = false, (gogoproto.moretags) = "yaml:\"packet_id\""];
+  repeated PacketFee packet_fees = 2 [(gogoproto.moretags) = "yaml:\"packet_fees\"", (gogoproto.nullable) = false];
```

Comment:
should this be `PacketFees`?

Reply:
Good question. It could be `PacketFees` (non repeated), but `PacketFees` is just really a wrapper type for marshalling the fees as bytes to store in state. 
I figured using regular `repeated PacketFee` here just reduced having to reference an additional child if that makes sense. e.g. instead of `identifiedFees.PacketFees.PacketFees` we can just use `identifiedFees.PacketFees`

Reply:
I like this idea. I can take a look later today 👍

Reply:
I don't think this is possible. If I understood the suggestion correctly it would be to do away with `PacketFees` and have that defined as an alias in the Go code and use `(gogoproto.castrepeated)`:
```
type PacketFees []PacketFee
```

I believe we still need the proto defined type `PacketFees` as the type we store [here](https://github.com/cosmos/ibc-go/blob/damian/760-effecient-storage/modules/apps/29-fee/keeper/keeper.go#L361) must implement `codec.ProtoMarshaller` and using an alias type will not have the proto specific methods needed to marshal bytes.

Reply:
Yea that seems correct. Looks like the SDK always does some workaround to directly setting a slice of a type (splitting up the type into a key/value or adding some other value to the key to create mapping to each element in the slice)

I'm in favor of the current implementation :+1: